### PR TITLE
Hpmework22

### PR DESCRIPTION
**Pull request by Darina198412**

### CODE REVIEW

**Comment by Berichicko** [comment]
The code is not compiled! Please finish the homework!You got this problem, please correct it. 

image

**Comment by Berichicko** [comment]
**HINT**

```
@Test
public void studentBtn() throws InterruptedException {
    //Login
    loginPage.login("dsalina1984@gmail.com", "Qazxsw123@Qazxsw123@");
    // navigate student Btn
    Thread.sleep(1000);
    // click student Button
    WebElement btnStudent = getBasePage().getDriver().findElement(By.cssSelector(".view-profile .name "));
    btnStudent.click();

    // Find and clear input field
    WebElement inputField = getBasePage().getDriver().findElement(By.cssSelector("#inputProfileName  "));
    Thread.sleep(1000);
    inputField.sendKeys(Keys.CONTROL,"a", Keys.BACK_SPACE);

    // Enter new student name
    Thread.sleep(1000);
    inputField.sendKeys("Daria 4");
    Thread.sleep(1000);
    inputField.sendKeys(Keys.ENTER);
    Thread.sleep(2000);

    // Enter current password
    WebElement inputPasswordField = getBasePage().getDriver().findElement(By.cssSelector("input#inputProfileCurrentPassword"));
    Thread.sleep(1000);
    inputPasswordField.sendKeys(Keys.CONTROL,"a", Keys.BACK_SPACE);
    Thread.sleep(1000);
    inputPasswordField.sendKeys("Qazxsw123@Qazxsw123@");
    Thread.sleep(1000);
    inputPasswordField.sendKeys(Keys.ENTER);

    // Click save Button
    WebElement saveBtn = getBasePage().getDriver().findElement(By.cssSelector(".btn-submit"));
    saveBtn.click();
}
```

1. Логин в систему.
2. Переход к кнопке "Student".
3. Клик на кнопку "Student".
4. Нахождение и очистка поля ввода.
5. Ввод нового имени студента.
6. Ввод текущего пароля.
7. Клик на кнопку "Save".

**Comment by Darina198412** [comment]
Александр я исправила домашку посмотри пожалуйста правильно ли я сделала assert, спасибо)))

**Comment by Berichicko** [comment]
Hello,

I have checked your homework 23, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

804deb64dc406a20a75da05d91a1a69e

image

**Comment by Berichicko** [comment]
Hello,

I have checked your homework 23, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

804deb64dc406a20a75da05d91a1a69e

image

File: src/test/java/pages/PlayListPage.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+public class PlayListPage extends BasePage {
+    LoginPage loginPage = new LoginPage();
+
+    // Locators create New PlayList
+    //navigate PlayList
+    By createPlayListBtn = By.cssSelector("[title='Create a new playlist'] ");
+    // click playList
+    By clickPlayList = By.cssSelector("[title='Create a new playlist']");
+      // navigate New playList
```
Comment (Berichicko) [inline src/test/java/pages/PlayListPage.java:18]: How can I beautify code in IntelliJ:

The formatting shortcuts in Intellij IDEA are :
For Windows : Ctrl + Alt + L.
For Ubuntu : Ctrl + Alt + Windows + L.
For Mac : ⌥ (Option) + ⌘ (Command) + L.

image

File: src/test/java/pages/PlayListPage.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        Assert.assertTrue(newPlayListName.isDisplayed());
+        return true;
+    }
+    // Play songs test
+    public void clickAllSongsBtn(){
+        WebElement allSongsBtn = driver.findElement(By.xpath("//nav[@id='sidebar']/section[@class='music']//a[@href='#!/songs']"));
+        allSongsBtn.click();
+    }
+    public void doubleClickSongBtn(){
+        WebElement songBtn = driver.findElement(By.cssSelector(".all-songs [draggable='true']:nth-of-type(6) .title "));
```
Comment (Berichicko) [inline src/test/java/pages/PlayListPage.java:217]: We can make the xpath or css the class variable (Locator)

File: src/test/java/pages/PlayListPage.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        WebElement playLInputField = driver.findElement(By.cssSelector("input[name='name'] "));
+        Thread.sleep(2000);
+        playLInputField.sendKeys(Keys.CONTROL,"a", Keys.BACK_SPACE);
+        Thread.sleep(1000);
+        playLInputField.sendKeys("King41");
+        playLInputField.sendKeys(Keys.ENTER);
+        Thread.sleep(1000);
+    }
+    public boolean isDisplayedNewPL(){
+        WebElement newPlayListName = driver.findElement(By.xpath("//li[@class='playlist playlist'][4]"));
```
Comment (Berichicko) [inline src/test/java/pages/PlayListPage.java:207]: We can make the xpath or css the class variable (Locator)

File: src/test/java/pages/PlayListPage.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        allSongsBtn.click();
+    }
+    public void doubleClickSongBtn(){
+        WebElement songBtn = driver.findElement(By.cssSelector(".all-songs [draggable='true']:nth-of-type(6) .title "));
+        new Actions(driver)
+                .doubleClick(songBtn)
+                .perform();
+    }
+    public boolean asssertEqualizerPlaying(){
+        WebElement equalizerBtn = driver.findElement(By.cssSelector("[alt='Sound bars']"));
```
Comment (Berichicko) [inline src/test/java/pages/PlayListPage.java:223]: We can make the xpath or css the class variable (Locator)

File: src/test/java/pages/PlayListPage.java [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        WebElement navigateFieldCreatePlayList = driver.findElement(navigateFildNewPlayL);
+        new Actions(driver)
+                .contextClick(navigateFieldCreatePlayList)
+                .perform();
+    }
+    public void newNamePlayList() throws InterruptedException {
+        WebElement addplayListNameField = driver.findElement(createNewName);
+        addplayListNameField.click();
+        addplayListNameField.clear();
+        Thread.sleep(1000);
```
Comment (Berichicko) [inline src/test/java/pages/PlayListPage.java:81]: Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

**Pull request closed without merging**